README.md

```diff
@@ -16,3 +16,86 @@ The project and Python dependencies can be installed from source with:
 ```
 pip3 install .
 ```
+
+## Configuration System
+
+This project provides a configuration system that supplies options (e.g., username, password) to a command-line interface (CLI) tool.
+The configuration system follows a tiered order, allowing options to be specified and overridden from both files and command-line options.
+
+### Configuration Sources
+
+In addition to CLI options, the configuration system loads options from [JSON](https://en.wikipedia.org/wiki/JSON) files located across multiple directories.
+By default, configuration files are named `edq-config.json`.
+This value is customizable, but this document will assume the default is used.
+
+For example, a configuration file containing the `user` and `pass` options might look like this:
+```json
+{
+    "user": "alice",
+    "pass": "password123"
+}
+```
+
+The table below summarizes the configuration sources in the order they are evaluated.
+The value from the later source in the table overrides the earlier one if there are multiple sources.
+
+| Source   | Description |
+| :-----   | :---------- |
+| Global   | Configuration that is loaded from a file in a user-specific location, which is platform-dependent. |
+| Local    | Configuration that is loaded from a file in the current or nearest ancestor directory. |
+| CLI File | Configuration that is loaded from one or more explicitly provided configuration files through the CLI. |
+| CLI      | Configuration that is loaded from the command line. |
+
+The system produces an error if a global or local configuration file is unreadable (but not missing), or if a CLI-specified file is unreadable or missing.
+
+#### Global Configuration
+
+The global configuration file defaults to `<platform-specific user configuration location>/edq-config.json`.
+The configuration location is chosen according to the [XDG standard](https://en.wikipedia.org/wiki/Freedesktop.org#Base_Directory_Specification) (implemented by [platformdirs](https://github.com/tox-dev/platformdirs)). 
+Below are examples of user-specific configuration file paths for different operating systems:
+ - Linux -- `/Users/<user>/.config/edq-config.json`
+ - Mac -- `/Users/<user>/Library/Application\ Support/edq-config.json`
```

Comment:
The space does not need to be escaped, since you are presenting it as an individual inline code block and not a command.